File: Stacks/ValidParentheses.java
package Stacks;

import java.util.ArrayList;
import java.util.Stack;

public class ValidParentheses {

    public boolean isValid(String s) {
        Stack<Character> chars = new Stack<Character>();

        ArrayList<String> validPairs = new ArrayList<String>();

        //Character init = s.charAt(0);

        //chars.push(init);

        validPairs.add("()");
        validPairs.add("{}");
        validPairs.add("[]");

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            System.out.println(c);
        }
        
        return false;
    }
    
}

File: Stacks/ValidParentheses.cpp
//
// Created by User on 3/4/2025.
//
#include <iostream>
#include <vector>
#include <stack>
using namespace std;

class ValidParentheses {
public:
    bool isValid(string s) {
        stack<char> chars;

        vector<string> validPairs = {"{}", "[]", "()"};

        for(int i = 0; i < s.length(); i++) {
            char c = s[i];
            if (chars.empty()) {
             chars.push(c);
            } else {
                char d = chars.top();
                string check = string(1, d) + string(1, c);
                if (validPairs.contains(check)) {
                  chars.pop();
                } else {
                  chars.push(c);
                  }

            }

        }
    }
};
